chain/consensus/ideal/ideal.go

```diff
@@ -0,0 +1,241 @@
+// Package ideal implements Ideal consensus for testing purposes only.
```

Comment:
Isn't ideal quite redundant with delegated? What about removing delegated and keeping ideal as the "debug" consensus (or vice-versa?). 

If we go this way I would maybe call it "debug" instead of "ideal" consensus. We would have to add cmds to be able to run it as part of the daemon, but I think this may be worth the effort.

Reply:
The idea of Ideal consensus is to have the simplest and most independent consensus protocol. The delegated consensus is more complex, at least its instantiation is not trivial, it can be spawned on several nodes. I would say that the delegation consensus is the next level: we start with the simplest reference model (ideal/debug consensus) and move forward with tests to be sure we test the target code (not consensus code) with ideal dependencies.

Reply:
That's all correct. I called its ideal following the Real/Ideal paradigm. No problems to rename it